src/main/java/codeu/controller/LoginServlet.java

```diff
@@ -60,26 +60,29 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
 
   /**
    * This function fires when a user submits the login form. It gets the username from the submitted
-   * form data, and then adds it to the session so we know the user is logged in.
+   * form data, checks that they're valid, and either adds the user to the session so we know the 
+   * user is logged in or shows an error to the user. 
    */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response)
       throws IOException, ServletException {
     String username = request.getParameter("username");
+    String password = request.getParameter("password");
 
-    if (!username.matches("[\\w*\\s*]*")) {
-      request.setAttribute("error", "Please enter only letters, numbers, and spaces.");
-      request.getRequestDispatcher("/WEB-INF/view/login.jsp").forward(request, response);
-      return;
+    if (userStore.isUserRegistered(username)) {
+      User user = userStore.getUser(username);
+      if (password.equals(user.getPassword())) {
+        request.getSession().setAttribute("user", username);
+        response.sendRedirect("/conversations");
+      }
+      else {
+        request.setAttribute("error", "Invalid password.");
+        request.getRequestDispatcher("/WEB-INF/view/login.jsp").forward(request, response);
+      }
     }
-
-    if (!userStore.isUserRegistered(username)) {
-      // TODO: add password
-      User user = new User(UUID.randomUUID(), username, null, Instant.now());
-      userStore.addUser(user);
+    else {
+      request.setAttribute("error", "Invalid password.");
```

Comment:
Should this be invalid username or password?

Reply:
ah yes! it should be username my bad